### PR TITLE
Remove unused code in PeerConnections.h

### PR DESCRIPTION
**Pull request by kghost**
#### Problem
Clean up code which is not used anymore

#### Change overview
* Remove APIs which use address as key.
* NodeId should be mandatory for a secure session

#### Testing
Verified using unit-tests

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

File: src/transport/PeerConnections.h [hunk 1]
```diff
@@ -114,11 +72,7 @@ class PeerConnections
                 mStates[i].SetPeerKeyID(peerKeyId);
                 mStates[i].SetLocalKeyID(localKeyId);
                 mStates[i].SetLastActivityTimeMs(mTimeSource.GetCurrentMonotonicTimeMs());
-
-                if (peerNode.ValueOr(kUndefinedNodeId) != kUndefinedNodeId)
-                {
-                    mStates[i].SetPeerNodeId(peerNode.Value());
-                }
+                mStates[i].SetPeerNodeId(peerNode);
```
Comment (bzbarsky-apple) [inline src/transport/PeerConnections.h:75]: What if caller passes `kUndefinedNodeId` ?

File: src/transport/PeerConnections.h [hunk 2]
```diff
@@ -225,14 +147,10 @@ class PeerConnections
             {
                 continue;
             }
-            if (peerKeyId == kAnyKeyId || iter->GetPeerKeyID() == peerKeyId)
+            if (iter->GetPeerNodeId() == nodeId && (peerKeyId == kAnyKeyId || iter->GetPeerKeyID() == peerKeyId))
```
Comment (bzbarsky-apple) [inline src/transport/PeerConnections.h:150]: This changes the handling around `kUndefinedNodeId`.  Is that expected/ok?

File: src/transport/PeerConnections.h [hunk 3]
```diff
@@ -303,14 +219,10 @@ class PeerConnections
             {
                 continue;
             }
-            if (iter->GetLocalKeyID() == localKeyId)
+            if (iter->GetPeerNodeId() == nodeId && iter->GetLocalKeyID() == localKeyId)
```
Comment (bzbarsky-apple) [inline src/transport/PeerConnections.h:222]: Again, changes handling of `kUndefinedNodeId`.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

**Comment by kghost** [comment]
I'd probably close this PR and move the content into #8206 , because most code introduced by this PR is immediately remove by that PR.

As @andy31415 suggested, I'll use `PeerId` class to remove (`NodeId`, `AdminId`) tuple.

**Pull request closed without merging**